### ISSUE TITLE
Title: Gemma 4 models: NaN logits (26B A4B) and infinite hang (E4B) on specific prompts

### ISSUE THREAD

**Issue by lknight** [EricLBuehler/mistral.rs#2051]
Describe the bug:

  Both Gemma 4 variants produce broken output on certain prompts:

  - google/gemma-4-26B-A4B-it returns Invalid sampling probability at index 0: NaN. The model likely produced NaN/Inf
  logits. as an SSE text chunk (not a JSON error), then immediately sends [DONE]. The client receives no usable content.
  - mistralrs-community/gemma-4-E4B-it-UQFF hangs indefinitely — sends only empty SSE keepalive lines (: \n\n) with no
  content tokens, never completes. Non-streaming also times out.

  Minimal reproducer (26B A4B, BF16):
  curl http://localhost:8142/v1/chat/completions \
    -H "Content-Type: application/json" \
    -d '{
      "model": "default",
      "messages": [
        {"role": "system", "content": "CRITICAL: Always reply in the language the user writes in.\n\nYou are a highly
  knowledgeable expert assistant. Before answering, reason through the problem step by step using your thinking
  capabilities.\n\n## Core rules\n- Always respond in the same language the user writes in.\n- Provide complete, concrete
  answers with specific details.\n- Structure your reasoning: break complex tasks into steps, then give a clear final
  answer."},
        {"role": "user", "content": "Could you start by asking me about instances when I procrastinate the most and then
  give me some suggestions to overcome it?"}
      ],
      "max_tokens": 2000,
      "stream": true
    }'

  Output: data: Invalid sampling probability at index 0: NaN. The model likely produced NaN/Inf logits.

  Same reproducer works for E4B (hangs instead of NaN).

  Notes:
  - Without the system prompt, BF16 weights do not reproduce the issue
  - UQFF quantization reproduces NaN even without system prompt
  - Rephrased prompts work fine ("What are the best tips to stop procrastinating?")
  - Simple prompts work fine ("Say hello")
  - Both streaming and non-streaming fail on the reproducer prompt
  - The NaN is returned as a plain-text SSE chunk, not a JSON error object — this breaks OpenAI-compatible clients (e.g.
  LiteLLM parses it as Expecting value: line 1 column 1)
  - Both models produce all output in reasoning_content field with content: null during streaming, which may be related

  Hardware / launch configs:

  26B A4B (BF16):
  - GPU: NVIDIA RTX PRO 6000 Blackwell Max-Q 96GB
  - Launch: mistralrs serve -m google/gemma-4-26B-A4B-it --pa-memory-fraction 0.4 --pa-cache-type f8e4m3 --max-seqs 16
  --port 8142 --paged-attn off

  E4B (UQFF):
  - GPU: NVIDIA RTX A6000 Ampere 48GB
  - Launch: mistralrs serve -m mistralrs-community/gemma-4-E4B-it-UQFF --port 8141

  Both models fail on the same prompt with different symptoms, suggesting the issue is in mistral.rs inference logic rather
  than hardware/driver specific.

  Latest commit or version: latest release (installed 2026-04-03)
